contracts/GatewayComposer.sol

```diff
@@ -68,6 +71,15 @@ contract GatewayComposer {
         _;
     }
 
+    /** Checks that mutex is acquired or not. */
+    modifier isMutexAcquired() {
```

Comment:
the full Mutex can be inside the modifier; eg

```
modifier Mutex() {
  require(
    mutexAcquired = false,
    "Mutex is already acquired."
  );
  mutexAcquired = true;
  _;
  mutexAcquired = false;
}
```

Reply:
Done!